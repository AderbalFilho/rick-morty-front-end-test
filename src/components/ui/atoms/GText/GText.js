import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './GText.scss';

function GText({ color, children, classes, family, ...rest }) {
  const className = classNames(
    'g-text',
    `g-text--${color}`,
    `g-text--${family}`,
    classes
  );

  return (
    <p className={className} {...rest}>
      {children}
    </p>
  );
}

GText.defaultProps = {
  color: 'white',
  family: 'nunito',
};

GText.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  color: PropTypes.oneOf(['darkest-grey', 'light-grey', 'white']),
  family: PropTypes.oneOf(['nunito', 'seravek']),
};

export default GText;
