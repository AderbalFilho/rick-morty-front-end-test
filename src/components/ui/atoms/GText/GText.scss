@import 'styles/mixins';

.g-text {
  @include base-text;

  margin: 0;
  padding: 0;
}

// Color
.g-text--darkest-grey {
  color: var(--color-darkest-grey);
}

.g-text--light-grey {
  color: var(--color-light-grey);
}

.g-text--white {
  color: var(--color-white);
}

// Family
.g-text--nunito {
  font-family: 'Nunito';
}

.g-text--seravek {
  font-family: 'Seravek';
}
