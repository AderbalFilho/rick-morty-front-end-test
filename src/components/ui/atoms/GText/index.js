import GText from './GText';

export default GText;
