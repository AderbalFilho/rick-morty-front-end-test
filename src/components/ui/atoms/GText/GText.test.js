import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import GText from './GText';

describe('GText', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<GText>inGaia</GText>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  describe('change color', () => {
    it('renders white when no color is passed', () => {
      const { container } = render(<GText>inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--white');
    });

    it('renders darkest-grey', () => {
      const { container } = render(<GText color="darkest-grey">inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--darkest-grey');
    });

    it('renders light-grey', () => {
      const { container } = render(<GText color="light-grey">inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--light-grey');
    });

    it('renders white', () => {
      const { container } = render(<GText color="white">inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--white');
    });
  });

  describe('change family', () => {
    it('renders Nunito when no family is passed', () => {
      const { container } = render(<GText>inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--nunito');
    });

    it('renders Seravek', () => {
      const { container } = render(<GText family="seravek">inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--seravek');
    });

    it('renders Nunito', () => {
      const { container } = render(<GText family="nunito">inGaia</GText>);
      const text = container.querySelector('.g-text');

      expect(text.classList).toContain('g-text--nunito');
    });
  });

  it('renders children', () => {
    const { container } = render(<GText>inGaia</GText>);
    const text = container.querySelector('.g-text');

    expect(text.textContent).toBe('inGaia');
  });

  it('renders classes', () => {
    const { container } = render(<GText classes="rick-text">inGaia</GText>);
    const text = container.querySelector('.g-text');

    expect(text.classList).toContain('rick-text');
  });
});
