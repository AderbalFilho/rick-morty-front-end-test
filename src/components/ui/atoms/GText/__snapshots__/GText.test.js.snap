// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GText matches snapshot 1`] = `
<p
  className="g-text g-text--white g-text--nunito"
>
  inGaia
</p>
`;
